src/legacy/core_plugins/visualizations/public/np_ready/public/embeddable/visualize_embeddable_factory.tsx

```diff
@@ -93,31 +93,31 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory<
     });
   }
 
-  public async createFromObject(
-    savedObject: VisSavedObject,
+  public createFromObject(
+    vis: Vis,
     input: Partial<VisualizeInput> & { id: string },
     parent?: Container
-  ): Promise<VisualizeEmbeddable | ErrorEmbeddable | DisabledLabEmbeddable> {
+  ): VisualizeEmbeddable | ErrorEmbeddable | DisabledLabEmbeddable {
     const savedVisualizations = getSavedVisualizationsLoader();
 
     try {
-      const visId = savedObject.id as string;
+      const visId = input.id as string;
```

Comment:
With this change, this `input.id` is actually the embeddable panel ID, not the saved vis object ID, which contributes to at least one of the functional test failures where adding a dashboard vis panel and then attempting to edit it fails to find the SO (because the URL on L107 is incorrect)

The correct saved vis object ID is there in `createFromSavedObject()`, but is not preserved anywhere inside `Vis` or when the saved object is converted to a serialized vis. So either we need to update the args to `createFromObject` to pass this in, or preserve the ID somehow in `ISavedVis`  and `SerializedVis`.

Reply:
thanks!